daemon/database/models/SwapIn.go

```diff
@@ -0,0 +1,42 @@
+package models
+
+import (
+	"time"
+)
+
+type SwapIn struct {
+	ID uint `gorm:"primaryKey;autoIncrement"`
+
+	// The swap identifier for the swap service
+	SwapID      string     `gorm:"not null"`
+	AmountSATS  uint64     `gorm:"not null"`
+	Status      SwapStatus `gorm:"type:status_enum;not null"`
+	SourceChain Chain      `gorm:"type:chain_enum;not null"`
+
+	// The address where we will pay to
+	DestinationAddress string
+	DestinationTx      string
+	// If we surpass the timeout block height we can ask for a refund
+	TimeoutBlockHeight uint64
+	// The address where the service will refund us to
+	RefundAddress    string
+	RefundTx         string
+	RefundPrivatekey string `gorm:"not null"`
+
+	// The redeem script for the on-chain transaction
+	RedeemScript string
+
+	// The lightning address where the money will be received
+	PaymentRequest string `gorm:"not null"`
+	PreImage       string
```

Comment:
Shall we use https://pkg.go.dev/github.com/lightningnetwork/lnd/lntypes#Preimage ?

Reply:
i would prefer to see the hex in the database for easy checking

Reply:
maybe we can use the type while working on it but make it automatically serialize/unserialize when it goes to/back from the DB?

Reply:
Yes go to/from hex is nice, the type is really useful

Reply:
done